components/WiFiComponent/WiFiConfig.c

```diff
@@ -239,23 +243,45 @@ void WifiConnectionTask()
     ESP_LOGI(TAG, "NVS init");
     ESP_ERROR_CHECK(nvs_flash_init());
     ESP_ERROR_CHECK(esp_netif_init());
-    SpiffsInit();
+    ExitFromApModeSemaphore = xSemaphoreCreateBinary();
+    if (xSemaphoreTake(HaveSaveForWifiSemaphore, 1) == pdTRUE)
+    {
+        ESP_LOGI(TAG, "we have save file ! ");
+        ReadFileFromSpiffsWithTxtFormat(WifiConfigAddressInSpiffs, "SSID", UserWifi.SSID, "PASS", UserWifi.PassWord, NULL, NULL);
+        HaveSaveFlag = 1;
+        WifiStationMode(UserWifi.SSID, UserWifi.PassWord);
+        if (xSemaphoreTake(ExitFromApModeSemaphore, 30 * 1000 / portTICK_PERIOD_MS) == pdTRUE)
+        {
+            ESP_LOGI(TAG, "FinishWifiConfig semaphore ");
+            vTaskDelay(3 * 1000 / portTICK_PERIOD_MS);
+            xSemaphoreGive(FinishWifiConfig);
+            vTaskDelete(NULL);
+        }
+        else
+        {
+            ESP_LOGE(TAG, "SSID or Password or wifi have problem !");
+        }
+    }
+    else
+    {
+        ESP_LOGE(TAG, "we does not have save file ! ");
+    }
     ESP_LOGI(TAG, "init softAP");
     ESP_ERROR_CHECK(WifiSoftAccessPointMode(ESP_WIFI_SSID, ESP_WIFI_PASS));
     StartMDNSServiceForWifi();
     server_ = StartWebServerLocally();
     ForFirstTimeFlag = 1;
     WaitSemaphore = xSemaphoreCreateBinary();
-    ExitFromApModeSemaphore = xSemaphoreCreateBinary();
     StayInApModeSemaphore = xSemaphoreCreateBinary();
     while (1)
     {
         ESP_LOGI(TAG, "wait \n");
         if (xSemaphoreTake(WaitSemaphore, portMAX_DELAY) == pdTRUE)
         {
+            ESP_LOGI(TAG, "give ssid and password but , we do'nt test them");
             vTaskDelay(3000 / portTICK_PERIOD_MS);
             WifiStationMode(UserWifi.SSID, UserWifi.PassWord);
-            vTaskDelay(5000 / portTICK_PERIOD_MS);
+            vTaskDelay(10 * 1000 / portTICK_PERIOD_MS);
```

Comment:
no magic number, use MACRO :)

Reply:
I define macro for this :"#define Sec 1000
#define Hour 3600"